hangupsbot/plugins/telesync/__init__.py

```diff
@@ -1035,27 +1069,42 @@ def syncprofile(bot, event, *args):
 
     parameters = list(args)
 
-    ho2tg_dict = bot.memory.get_by_path(['profilesync'])['ho2tg']
-    tg2ho_dict = bot.memory.get_by_path(['profilesync'])['tg2ho']
-
     if len(parameters) != 1:
         yield from bot.coro_send_message(event.conv_id, "supply registration id as single parameter")
 
     else:
-        registration_id = str(parameters[0])
-        if registration_id in ho2tg_dict:
-            ho_id = registration_id
-            tg_id = ho2tg_dict[registration_id]
+        registration_code = str(parameters[0])
+        ho2tg_dict = bot.memory.get_by_path(['profilesync'])['ho2tg']
+        tg2ho_dict = bot.memory.get_by_path(['profilesync'])['tg2ho']
+
+        hangouts_uid = str(event.user_id.chat_id)
+
+        if( hangouts_uid in ho2tg_dict
+                and ho2tg_dict[hangouts_uid] in tg2ho_dict
+                and not isinstance(tg2ho_dict[ho2tg_dict[hangouts_uid]], str) ):
+
+            yield from bot.coro_send_message(
+                event.conv_id,
+                "profile is already synced" )
+
+        elif not registration_code.startswith(reg_code_prefix):
+            yield from bot.coro_send_message(
+                event.conv_id,
+                "execute /syncprofile command in a private chat with the bot on telegram first" )
+
+        elif registration_code in ho2tg_dict:
+            ho_id = registration_code
+            telegram_uid = str(ho2tg_dict[registration_code])
 
-            chat_id = event.user_id.chat_id
-            user_gplus = 'https://plus.google.com/u/0/{}/about'.format(chat_id)
+            user_gplus = 'https://plus.google.com/u/0/{}/about'.format(hangouts_uid)
 
-            tg2ho_dict[tg_id] = { 'ho_id': ho_id,
-                                  'chat_id': chat_id,
-                                  'user_gplus': user_gplus }
+            tg2ho_dict[telegram_uid] = {
+                'registration_code': registration_code,
+                'chat_id': hangouts_uid,
+                'user_gplus': user_gplus }
+            ho2tg_dict[hangouts_uid] = telegram_uid
 
-            # del ho2tg_dict[str(registration_id)]
-            ho2tg_dict[str(event.user_id.chat_id)] = str(tg_id)
+            del ho2tg_dict[registration_code]
```

Comment:
old telesync never removed previous registration codes, this was probably made even more difficult because the registration code was permanently stored as a reference to a hangouts profile. we make it non-resuseable after first successful use, but store it in the dict just for record-keeping purposes